util/io/src/worker.rs

```diff
@@ -91,8 +92,9 @@ impl Worker {
 
 			while !deleting.load(AtomicOrdering::Acquire) {
 				match stealer.steal() {
-					chase_lev::Steal::Data(work) => Worker::do_work(work, channel.clone()),
-					_ => break,
+					deque::Steal::Data(work) => Worker::do_work(work, channel.clone()),
+					deque::Steal::Retry => {},
```

Comment:
this was a bug, if worker lost the race for stealing data, worker was incorrectly killed

Reply:
the break is from the inner `while` loop

Reply:
I'm pretty sure it's going to increase CPU usage, but should also improve performance, cause previously if a thread lost the race it would go to sleep vs re-trying and get the work done. If it went to sleep other threads had to finish whatever they were doing and then pick up the new task.

Reply:
there is a case when `Worker` is being dropped. If `steal` returns `Retry`, work will never be resumed